pkg/nvmeclient/nvme_client.go

```diff
@@ -558,10 +558,13 @@ func ConnectAllNVMEDevices(logPageEntries []*hostapi.NvmeDiscPageEntry, hostnqn
 						// on the discovery-service or the DS is down on that node but the IO controller is still accessible.
 						logrus.WithError(perr).Warn("failed to connect IO controller. This may be a transient error or due to a node being down.",
 							"Continuing to attempt connection until the discovery-service stops providing the down node's address..")
+						continue
 					}
 				}
+			} else {
+				logrus.WithError(err).Errorf("failed to connect to: %s", request.Traddr)
+				continue
 			}
-			continue
```

Comment:
I'm not sure I follow the logic here. If `connectNVMEDevicesWithRetry` succeeds, won't we just move to the next log page? Where do we add the ctrl to the list in this case?
(In general, this code makes my head hurt -- surely we can make it simpler to follow).